### PR TITLE
[CPM-123] Prefill UoA Course

### PR DESCRIPTION
**Pull request by FocalChord**
## Changes

- Backend has changed to return the required information for a course
- Frontend has a button which asks to prefill the Course information when the course code regex matches
- Frontend hooked up with backend

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

Addresses [CPM-123]

[CPM-123]: https://momentum-agile.atlassian.net/browse/CPM-123

### CODE REVIEW

File: client/src/pages/Courses/useCourses.js [hunk 1]
```diff
@@ -53,12 +53,16 @@ const useCourses = () => {
             .catch((e) => console.log(e));
     };
 
+    const prefillCourse = (subject, courseNumber) => {
```
Comment (vee-christella) [inline client/src/pages/Courses/useCourses.js:56]: Can we add a comment to note that this reaches the university API?

**Review by vee-christella — APPROVED** [review]
LGTM for the design aspects. I think we should change the wording of `Subject` to something like `Degree` or `Department` because I feel like  Subject is a bit vague and isn't a term we use much at university (is it?)
image

**Comment by vee-christella** [comment]
Additionally, let's create another ticket to notify the user that their courses may be overriden on prefilling courses if the same course code exists.

**Review by vee-christella — APPROVED** [review]
Nice!

**Review by Richardng12 — APPROVED** [review]
Amazing work by @FocalChord

**Pull request merged**